### PR TITLE
Bugfix/KITOSUDV-3925

### PR DESCRIPTION
**Pull request by a-naskret**
## KITOS Pull request template
The following procedure dictates the steps needed before a Pull request can be merged into master.

- [x] **Implement**: 
      *All requirements are implemented and unit tests are green*
      
- [x] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [x] **Green on integration**: 
      *All integration tests are green on integration*
      
- [x] **Request review**: 
      *Tag whomever you wish to review your code*
      
- [ ] **Review completed**: 
      *Reviewer ticks this box when review comments have been submitted*
      
- [ ] **Changes**: 
      *PR owner and reviewer agrees on which changes must be made and the changes are committed.*
      
- [ ] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [ ] **Green on integration**: 
      *All integration tests are green on integration*

### CODE REVIEW

File: Presentation.Web/Swagger/SwashbuckleExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
+﻿using Swashbuckle.Swagger;
+using System.Collections.Generic;
+
```
Comment (mrjsawdk) [inline Presentation.Web/Swagger/SwashbuckleExtensions.cs:3]: We already have an integration test that covers the V1 and V2 json files.
Extend that suite with two tests that verify that a v2 dto is not present in the v1 json but IS in the v1 json and vice versa.

**Review by mrjsawdk — COMMENTED** [review]
@a-naskret Lookin' good.
One question and one missing validation

File: Presentation.Web/Swagger/SwashbuckleExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            }
+        }
+
+        public static IEnumerable<Schema> EnumerateSchema(this Schema schema, IDictionary<string, Schema> listOfDefinition, int dept = 0)
+        {
+            if (schema == null)
+            {
+                yield break;
+            }
+            if (dept > 3) //if the depth is too big, the api v1 generation won't work
```
[the remark this answers is not in the document]
Comment (a-naskret) [reply Presentation.Web/Swagger/SwashbuckleExtensions.cs:77]: there's an infinite loop, or it takes a really long time to process, when depth was set to 10 I entered the /swagger/docs/1 while it was generating I went to grab a coffee, and it was still processing the children

[the remark this answers is not in the document]
Comment (a-naskret) [reply Presentation.Web/Swagger/SwashbuckleExtensions.cs:77]: I looked through the (old) swagger.json, and I haven't seen a type with depth greater than 2, but I can try increasing the depth, and check if it works

[the remark this answers is not in the document]
Comment (mrjsawdk) [reply Presentation.Web/Swagger/SwashbuckleExtensions.cs:77]: that's because you are not using loop detection :-)
Solve the problem the right way ;)

**Pull request closed without merging**